drivers/scale_host.go

```diff
@@ -121,39 +144,53 @@ func (s *ScaleHostDriver) Execute(conf interface{}, apiClient *client.RancherCli
 	hostSelectorPresent := false
 	baseHostIndex = -1
 	for _, host := range hostCollection.Data {
-		labels := host.Labels
-		labelFound := false
-		for k, v := range labels {
-			if !strings.EqualFold(k, key) {
-				continue
+		if config.HostTemplateID != "" {
+			// add hosts with specified hostTemplateId
+			if host.HostTemplateId == config.HostTemplateID {
+				hostScalingGroup = append(hostScalingGroup, host)
+				if host.Driver != "" {
+					baseHostIndex = int64(len(hostScalingGroup)) - 1
+				}
 			}
-			if !strings.EqualFold(v.(string), value) {
-				continue
+		} else {
+			labels := host.Labels
+			labelFound := false
+			for k, v := range labels {
+				if !strings.EqualFold(k, key) {
+					continue
+				}
+				if !strings.EqualFold(v.(string), value) {
+					continue
+				}
+				labelFound = true
+				break
 			}
-			labelFound = true
-			break
-		}
 
-		if !labelFound {
-			continue
-		}
+			if !labelFound {
+				continue
+			}
 
-		if host.State == "error" {
-			continue
-		}
+			if host.State == "error" {
+				continue
+			}
 
-		hostSelectorPresent = true
-		hostScalingGroup = append(hostScalingGroup, host)
+			hostSelectorPresent = true
+			hostScalingGroup = append(hostScalingGroup, host)
 
-		if host.Driver != "" {
-			baseHostIndex++
+			if host.Driver != "" {
+				baseHostIndex = int64(len(hostScalingGroup)) - 1
+			}
 		}
 	}
 
-	if hostSelectorPresent == false {
+	if config.HostTemplateID == "" && hostSelectorPresent == false {
 		return http.StatusBadRequest, fmt.Errorf("No host with label %v exists", hostSelector)
 	}
 
+	if baseHostIndex == -1 && config.HostTemplateID != "" {
```

Comment:
not fully get your idea, the basehostIndex here is to help filter out host with specified labels or hostTemplateId. 
Since we sort hosts by filters["sort"] = "created" and filters["order"] = "desc", with basehostIndex we can get the first host being created that with specified labels or hostTemplateId. And according to that baseHost name to naming new hosts. Thus, I do not think we can name like hardcoded